wfe/wfe.go

```diff
@@ -499,64 +502,80 @@ func (wfe *WebFrontEndImpl) verifyPOST(
 	ctx context.Context,
 	logEvent *requestEvent,
 	request *http.Request,
-	kx keyExtractor) ([]byte, *jose.JSONWebKey, *acme.ProblemDetails) {
+	kx keyExtractor) ([]byte, string, *jose.JSONWebKey, *acme.ProblemDetails) {
 
 	if prob := wfe.validPOST(request); prob != nil {
-		return nil, nil, prob
+		return nil, "", nil, prob
 	}
 
 	if request.Body == nil {
-		return nil, nil, acme.MalformedProblem("no body on POST")
+		return nil, "", nil, acme.MalformedProblem("no body on POST")
 	}
 
 	bodyBytes, err := ioutil.ReadAll(request.Body)
 	if err != nil {
-		return nil, nil, acme.InternalErrorProblem("unable to read request body")
+		return nil, "", nil, acme.InternalErrorProblem("unable to read request body")
 	}
 
 	body := string(bodyBytes)
 	parsedJWS, err := wfe.parseJWS(body)
 	if err != nil {
-		return nil, nil, acme.MalformedProblem(err.Error())
+		return nil, "", nil, acme.MalformedProblem(err.Error())
 	}
 
 	pubKey, prob := kx(request, parsedJWS)
 	if prob != nil {
-		return nil, nil, prob
+		return nil, "", nil, prob
 	}
 
 	return wfe.verifyJWS(pubKey, parsedJWS, request)
 }
 
-func (wfe *WebFrontEndImpl) verifyJWS(
+// Checks parsed JWS whether it matches the given public key, does some general
+// checks, and extracts the URL header parameter.
+func (wfe *WebFrontEndImpl) verifyJWSLight(
```

Comment:
Sorry, I think I forgot to hit submit to include this comment in my last review:

Can we brainstorm a new name for this function? What about splitting it up into two functions named `verifyJWSSignature` and `extractJWSURL`?  I think that will be more descriptive than `verifyJWSLight`.

Reply:
Yes, please :) I tentatively named it `verifyJWSLight` while I wasn't sure what exactly will end up in it, and since it is a subset of what `verifyJWS` is doing ;-)

Reply:
I think I'll name the first one `verifyJWSSignatureAndAlgorithm` (at least for now), as a reminder that the code for checking the algorithm is still missing ;-)

Reply:
I did the refactoring in 15befc4c28430080ad0ef5dc8d825b54f48cf529. I think it's now much better than before.